src/messages.ts

```diff
@@ -543,20 +543,90 @@ export class Messages<T extends string> {
     exitCodeOrCause?: number | Error,
     cause?: Error
   ): SfdxError {
+    return this.createSfdxError('error', key, tokens, actionTokens, exitCodeOrCause, cause);
+  }
+
+  /**
+   * Convenience method to create warning using message labels.
+   *
+   * `warning.name` will be the upper-cased key, remove prefixed `warning.` and will always end in Warning.
+   * `warning.actions` will be loaded using `${key}.actions` if available.
+   *
+   * @param key The key of the warning message.
+   * @param tokens The warning message tokens.
+   * @param actionTokens The action messages tokens.
+   * @param codeOrCause The error that caused this warning.
+   * @param cause The underlying error that caused this warning to be created.
+   */
+  public createWarning(
+    key: T,
+    tokens: Tokens = [],
+    actionTokens: Tokens = [],
+    codeOrCause?: Error,
+    cause?: Error
+  ): SfdxError {
+    return this.createSfdxError('warning', key, tokens, actionTokens, codeOrCause, cause);
+  }
+
+  /**
+   * Convenience method to create info using message labels.
+   *
+   * `info.name` will be the upper-cased key, remove prefixed `info.` and will always end in Info.
+   * `info.actions` will be loaded using `${key}.actions` if available.
+   *
+   * @param key The key of the warning message.
+   * @param tokens The warning message tokens.
+   * @param actionTokens The action messages tokens.
+   * @param codeOrCause The error that caused this warning.
+   * @param cause The underlying error that caused this warning to be created.
+   */
+  public createInfo(
```

Comment:
Why do `createInfo` and `createWarning` create `SfdxErrors`?

Reply:
SfdxError provides the data model needed to express actions. This seemed like the simplest way to accomplish this. I can refactor SfdxError a bit to extract bits needed by all three, but that would probably just be actions.

Reply:
Yeah, I think it might be worth putting in the time to abstract out the shared functionality - it feels slightly off to rely on an Error class for info and warning messages

Reply:
The sf-plugins-core change I amde to support info and warn is doing [something similar](https://github.com/salesforcecli/sf-plugins-core/blob/phale/custom-error-messages/src/sfCommand.ts#L225). Do I need to make a similar change there?

Reply:
The real benefit is getting more that a simple string message. The message.createWarning allows for the injection of message and actions, which is one of the components Matt wants for warning and info messages. And a warning could actually come from a caught error, so being able to use an error class as the basis for info/warning seemed reasonable to me.

Reply:
@mdonnalley your thoughts?

Reply:
@mdonnalley I went ahead and created a new type `StructuredMessage` and modified createInfo and createWarning to return that structure.